cluster/cluster.go

```diff
@@ -0,0 +1,373 @@
+				},
+				{
+					Name: "tenant-11",
+					Port: 9011,
+				},
+				{
+					Name: "tenant-12",
+					Port: 9012,
+				},
+				{
+					Name: "tenant-13",
+					Port: 9013,
+				},
+				{
+					Name: "tenant-14",
+					Port: 9014,
+				},
+				{
+					Name: "tenant-15",
+					Port: 9015,
+				},
+				{
+					Name: "tenant-16",
+					Port: 9016,
+				},
+			},
+			Selector: map[string]string{
+				"app": serviceName,
+			},
+			ClusterIP:                "None",
+			PublishNotReadyAddresses: true,
+		},
+	}
+
+	res, err := clientset.CoreV1().Services("default").Create(&scSvc);
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Println(res.String())
+
+}
+
+//Creates a the "secrets" of a tenant, for now it's just a plain configMap, but it should be upgraded to secret
+func CreateTenantSecret(tenantShortName string) {
+	config := getConfig()
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	secretsName := fmt.Sprintf("%s-env", tenantShortName)
+
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretsName,
+			Labels: map[string]string{
+				"app": tenantShortName,
+			},
+		},
+		Data: map[string]string{
+			"MINIO_ACCESS_KEY": "minio",
+			"MINIO_SECRET_KEY": "minio123",
+		},
+	}
+
+	res, err := clientset.CoreV1().ConfigMaps("default").Create(&configMap);
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Println(res.String())
+
+}
+
+//Creates a service that will resolve to any of the hosts within the storage cluster this tenant lives in
+func CreateTenantService(tenantName string, tenantPort int32, storageClusterNum string) {
+	config := getConfig()
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	scSvc := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tenantName,
+			Labels: map[string]string{
+				"name": tenantName,
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name: "http",
+					Port: tenantPort,
+				},
+			},
+			Selector: map[string]string{
+				"sc": fmt.Sprintf("storage-cluster-%s", storageClusterNum),
+			},
+		},
+	}
+
+	res, err := clientset.CoreV1().Services("default").Create(&scSvc);
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Println("done creating tenant service for tenant %s ")
+	fmt.Println(res.String())
+
+}
+
+type Tenant struct {
+	Name              string
+	Port              int32
+	StorageClusterNum string
+}
+
+func nodeNameForSCHostNum(storageClusterNum string, hostNum string) string {
+	switch hostNum {
+	case "1":
+		return "m3cluster-worker"
+	case "2":
+		return "m3cluster-worker2"
+	case "3":
+		return "m3cluster-worker3"
+	case "4":
+		return "m3cluster-worker4"
+	default:
+		return "m3cluster-worker"
+	}
+}
+
+func getDisks(storageClusterNum string, hostNum string) []string {
+	switch storageClusterNum {
+	case "2":
+		return []string{"/mnt/disk5", "/mnt/disk6", "/mnt/disk7", "/mnt/disk8"}
```

Comment:
Is the prefix here /mnt/disk going to be constant ?

Reply:
not really, this should be a smarter function that knows how many disks are available and return the proper array, I guess the prefix is up for grabs, what do you think is the best mount point for the disks?